tests/core/selection/optimization.js

```diff
@@ -124,6 +124,33 @@
 
 				assert.isFalse( editor._.shiftPressed );
 			} );
+		},
+
+		// (#3705)
+		'test deleting blocks': function( editor, bot ) {
+			bot.setData( '<p>&nbsp;</p><p>Whatever</p>', function() {
+				var range = editor.createRange(),
+					extractedHtml;
+
+				range.selectNodeContents( editor.editable() );
+				range.select();
+				extractedHtml = editor.extractSelectedHtml( true );
+
+				// In Safari the same selection has different range offsets and endContainer. Because of that it is automatically fixed
+				// and empty paragraph is converted to <br/> element.
+				if ( CKEDITOR.env.safari ) {
+					assert.isInnerHtmlMatching( '<br data-cke-eol="1" /><p>Whatever</p>', extractedHtml, 'Extracted HTML is incorrect.' );
+				} else {
+					assert.isInnerHtmlMatching( '<p>@</p><p>Whatever</p>', extractedHtml, 'Extracted HTML is incorrect.' );
+				}
+
+				// And in IE8 only node content is removed.
+				if ( CKEDITOR.env.ie && CKEDITOR.env.version == 8 ) {
+					assert.isInnerHtmlMatching( '<p>@@</p><p></p>', editor.editable().getHtml(), 'Editor content after extraction is incorrect.' );
```

Comment:
Have you checked how it worked before? From what I see (manual tests with jsfiddle provided in initial report) for both `4.12.0` and `4.13.0` `editor.editable().getHtml()` returns `<p>&nbsp;</p>` on IE8 :thinking:

But I also see that providing both `true` or `false` to `skipBogus` doesn't change anything here so it's not caused by this change, maybe it is something with focus behaviour in unit vs manual testing :thinking: Let's leave it like that then :ok_hand:

Reply:
And yes it was `true` before and now with `!CKEDITOR.env.webkit` is still `true` for IE8 for sure it's not this change. Still kind of confusing :thinking: